runtime/jvmti/jvmtiThread.c

```diff
@@ -1109,21 +1117,41 @@ resumeThread(J9VMThread *currentThread, jthread thread)
 		if (NULL != targetThread)
 #endif /* JAVA_SPEC_VERSION >= 19 */
 		{
+#if JAVA_SPEC_VERSION >= 19
+			if (0 == J9OBJECT_U32_LOAD(currentThread, threadObject, vm->isSuspendedInternalOffset)) {
+				rc = JVMTI_ERROR_THREAD_NOT_SUSPENDED;
+			} else {
+				J9OBJECT_U32_STORE(currentThread, threadObject, vm->isSuspendedInternalOffset, 0);
+			}
+#endif /* JAVA_SPEC_VERSION >= 19 */
+
+#if JAVA_SPEC_VERSION >= 19
+			if ((0 == J9OBJECT_U32_LOAD(currentThread, targetThread->threadObject, vm->isSuspendedInternalOffset))
+			&& (0 == J9OBJECT_U32_LOAD(currentThread, targetThread->carrierThreadObject, vm->isSuspendedInternalOffset)))
+			{
```

Comment:
Share common code from the else block. Also, the ifdefs can be optimized.
```suggestion
		if (NULL != targetThread)
#endif /* JAVA_SPEC_VERSION >= 19 */
		{
#if JAVA_SPEC_VERSION >= 19
			if ((0 == J9OBJECT_U32_LOAD(currentThread, targetThread->threadObject, vm->isSuspendedInternalOffset))
			&& (0 == J9OBJECT_U32_LOAD(currentThread, targetThread->carrierThreadObject, vm->isSuspendedInternalOffset)))
#endif /* JAVA_SPEC_VERSION >= 19 */
			{
```

Reply:
added ifdefs and else blocks are optimized as suggested.